heroku/heroku-vscode
# Added instructions when no apps are found in the git remotes

This PR provides the following functionality:

1. Displays a message when the project open does not have a heroku git remote.
<img width="461" alt="image" src="https://github.com/user-attachments/assets/be3b99d0-7de7-44c3-9a53-7edbd8f952e9">

2. Notifies the user when an app found in the git remote appears to be a Heroku app but is not listed as an app in Heroku
<img width="450" alt="image" src="https://github.com/user-attachments/assets/d1fa3373-96b7-4170-b28d-17f9e14c81ea">

3. Watches for changes to the `./.git/config` and updates the resource explorer when apps are added or removed from the git remote. e.g. when `heroku git:remote -a <MY_APP>` is executed.

## To test the "Getting started" message
1. Run the extension and open a sources project that does not have a Heroku app specified in the git remote.
2. Open the Heroku extension sidebar panel using the Heroku icon in the sidebar.
3. Observe the message notifying the user that the project does not appear to be a Heroku app.
 <img width="461" alt="image" src="https://github.com/user-attachments/assets/be3b99d0-7de7-44c3-9a53-7edbd8f952e9">

## To test the notification of a non-heroku app existing in the git remote.
1. With the same non-Heroku project open, add a git remote that looks like a Heroku app. `git remote add heroku.dev https://git.heroku.com/non-existent-app.git`
2. Observe the warning message in the lower right.
<img width="450" alt="image" src="https://github.com/user-attachments/assets/d1fa3373-96b7-4170-b28d-17f9e14c81ea">

## To test the git remote watch function
1. In a non-Heroku sources project and with the Heroku extension sidebar panel open, run `heroku apps:create` in the vscode terminal or an external terminal with the project root as the current working directory
2. Observe the Heroku extension panel is updated to reflect the new app
3. In the same terminal, run `git remote remove heroku` 
4. Observe the getting started message is once again displayed.
